filters/wasm/wasm.go

```diff
@@ -0,0 +1,104 @@
+package wasm
+
+import (
+	"context"
+	"net/url"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
+	"github.com/zalando/skipper/filters"
+)
+
+type wasmSpec struct{}
+
+type wasm struct {
+	code     []byte
+	mod      api.Module
+	request  api.Function
+	response api.Function
+}
+
+func NewWASM() filters.Spec {
+	return &wasmSpec{}
+}
+
+// Name implements filters.Spec.
+func (*wasmSpec) Name() string {
+	return filters.WASMName
+}
+
+// CreateFilter implements filters.Spec.
+func (*wasmSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
+	if len(args) != 1 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+	src, ok := args[0].(string)
+	if !ok {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+	u, err := url.Parse(src)
+	if err != nil {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	var code []byte
+
+	switch u.Scheme {
+	case "file":
+		code, err = os.ReadFile(u.Path)
```

Comment:
I think we should make this lazy and cacheable  and cleanup on filter close

Reply:
yes, as I wrote in the PR title super basic. I used https://github.com/tetratelabs/wazero/tree/main/examples/basic to test the super basic example.